src/Kernel/Magic/Resolution/MagicResolver.cs

```diff
@@ -142,7 +143,7 @@ public IEnumerable<MagicSymbol> FindMagic(AssemblyInfo assm)
                         .GetTypes()
                         .Where(t =>
                         {
-                            if (!t.IsClass && t.IsAbstract) { return false; }
+                            if (!t.IsClass || t.IsAbstract) { return false; }
```

Comment:
!

Reply:
Indeed, this was a problem! As it turns out, the net effect of this typo was to cause spurious errors to appear in debug logs as well as a slight degradation in the performance of magic command discovery.